### PR TITLE
Allow searching file by fileid

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes https://github.com/nextcloud/server/issues/4193

Note that multiple results can be returned for a single id since a user can have access to single file trough multiple paths

### CODE REVIEW

File: apps/dav/lib/Files/FileSearchBackend.php [hunk 1]
```diff
@@ -127,7 +127,7 @@ public function getPropertyDefinitionsForScope($href, $path) {
 			new SearchPropertyDefinition(FilesPlugin::OWNER_DISPLAY_NAME_PROPERTYNAME, false, true, false),
 			new SearchPropertyDefinition(FilesPlugin::DATA_FINGERPRINT_PROPERTYNAME, false, true, false),
 			new SearchPropertyDefinition(FilesPlugin::HAS_PREVIEW_PROPERTYNAME, false, true, false, SearchPropertyDefinition::DATATYPE_BOOLEAN),
-			new SearchPropertyDefinition(FilesPlugin::INTERNAL_FILEID_PROPERTYNAME, false, true, false, SearchPropertyDefinition::DATATYPE_NONNEGATIVE_INTEGER),
+			new SearchPropertyDefinition(FilesPlugin::INTERNAL_FILEID_PROPERTYNAME, true, true, false, SearchPropertyDefinition::DATATYPE_NONNEGATIVE_INTEGER),
```
Comment (nickvergessen) [inline apps/dav/lib/Files/FileSearchBackend.php]: please move up to `// queryable properties`

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: apps/dav/lib/Files/FileSearchBackend.php (hunk 1 above)
Comment (icewind1991) [inline apps/dav/lib/Files/FileSearchBackend.php]: done

**Comment by rullzer** [comment]
@tobiasKaminsky can you verify if this works for you?

**Review by nickvergessen — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Seems to do the job :+1:

**Comment by tobiasKaminsky** [comment]
Nice, thank you 👍

**Comment by MorrisJobke** [comment]
@icewind1991 @tobiasKaminsky Do you have an example for this and could maybe document this?

**Pull request merged**